nixos/modules/services/misc/nix-daemon.nix

```
@@ -704,6 +706,12 @@ in
             (map (m: m.hostName)
               (filter (badMachine) cfg.buildMachines)));
         }
+        {
+          assertion = any machineWithKey cfg.buildMachines -> isNixAtLeast "2.4pre";
+          message = ''
+            The nix.buildMachines.*.publicHostKey option only works with Nix >= 2.4.
+          '';
+        }
       ];
```

```suggestion
      ] ++ lib.optional (isNixAtLeast "2.4pre") {
        assertion = any machineWithKey cfg.buildMachines;
        message = ''
          The nix.buildMachines.*.publicHostKey option only works with Nix >= 2.4.
        '';
      };
```
or maybe an apply function?